src/main/java/uk/gov/hmcts/reform/unspec/callback/CallbackHandler.java

```diff
@@ -22,7 +22,17 @@ protected String callbackKey(CallbackType type) {
     }
 
     protected String callbackKey(CallbackType type, String pageId) {
-        return pageId == null ? type.getValue() : type.getValue() + "-" + pageId;
+        return callbackKey(null, type, pageId);
+    }
+
+    protected String callbackKey(CallbackVersion version, CallbackType type) {
+        return callbackKey(version, type, null);
+    }
+
+    protected String callbackKey(CallbackVersion version, CallbackType type, String pageId) {
+        String formattedVersion = ofNullable(version).map(x -> x.toString() + "-").orElse("");
+        String formattedPageId = ofNullable(pageId).map(x -> "-" + x).orElse("");
```

Comment:
can we have version and pageId instead of x?

Reply:
They're already defined in the scope of the method (the method params), any other ideas? Otherwise I think x is fine